modules/deployment scripts/aad-app/main.bicep

```diff
@@ -0,0 +1,68 @@
+@description('The location in which the deployment script will be ran')
+param location string
+
+@description('The display name of the AzureAD application')
+param displayName string
+
+@description('The reply application reply URLs')
+param replyUrls array = []
+
+@description('The URL to the application homepage')
+param appUri string
+
+@description('When true, allows the AzureAD application to issue access tokens (used for implicit flow)')
+param enableAccessTokenIssuance bool
+
+@description('When true, allows the AzureAD application to issue ID tokens (used for implicit & hybrid flows)')
+param enableIdTokenIssuance bool
+
+@description('The resource identifier of the managed identity used for the deployment script')
+param managedIdentityResourceId string
+
+@description('Microsoft Graph scope IDs to grant the AzureAD application')
+param microsoftGraphScopeIdsToGrant array = []
+
+@description('Microsoft Graph role IDs to grant the AzureAD application')
+param microsoftGraphAppRoleIdsToGrant array = []
+
+var name = 'AadAppScript'
```

Comment:
You'll need to make this name more unique, otherwise it won't be happy if you are deploying multiple AAD apps in the same deployment (I just discovered this with my aad-group module).

You could use the following to make it unique enough:
```bicep
var name = 'AadAppScript-${displayName}'
```